src/HttpClient/HttpClient.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\HttpClient;
+
+use Sentry\Client;
+use Sentry\Options;
+
+class HttpClient implements HttpClientInterface
+{
+    /**
+     * @var Options
+     */
+    protected $options;
+
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    protected $sdkIdentifier;
+
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    protected $sdkVersion;
+
+    public function __construct(Options $options, string $sdkIdentifier, string $sdkVersion)
+    {
+        $this->options = $options;
+        $this->sdkIdentifier = $sdkIdentifier;
+        $this->sdkVersion = $sdkVersion;
+    }
+
+    public function sendRequest(string $requestData): Response
+    {
+        $dsn = $this->options->getDsn();
+        if (null === $dsn) {
+            throw new \RuntimeException('The DSN option must be set to use the HttpClient.');
+        }
```

Comment:
What happens when we set the DSN to `null` to disable the client? This may be a big BC, for framework libraries too, we need to propagate this outside.

Reply:
An upcoming feature will require us always to construct an envelope in certain scenarios. I agree that this is not the best solution. I have to think about improving it.